### ISSUE TITLE
Update to OS 12.3 causing problems to VLC addon / VLC-Telnet integration

### ISSUE THREAD

**Issue by LeoCal** [home-assistant/operating-system#3373]
### Describe the issue you are experiencing

After updating to OS 12.3, I started facing issue with VLC addon / VLC Telnet integration.
More specifically, I get an intermittent and chopping sound, which has never occurred to me in several years using VLC.
It seems multiple users are facing a similar issue, as reported to VLC addon as well: https://github.com/home-assistant/addons/issues/3594 

Here's the details of my HA setup:
Core: 2024.5.3
Supervisor: 2024.05.1
Operating System: 12.3
Frontend: 20240501.1

### What operating system image do you use?

rpi4-64  (Raspberry Pi 4/400 64-bit OS)

### What version of Home Assistant Operating System is installed?

Home Assistant OS 12.3

### Did you upgrade the Operating System.

Yes

### Steps to reproduce the issue

1. Upgrade to OS 12.3
2. Try to reproduce a sound through the VLC addon with VLC Telnet integration
3. Notice how sound is chopping and intermittent

### Anything in the Supervisor logs that might be useful for us?

```txt
You can refer to the issue reported to the VLC addon for more logs from me and other users:
https://github.com/home-assistant/addons/issues/3594
```

### Anything in the Host logs that might be useful for us?

```txt
You can refer to the issue reported to the VLC addon for more logs from me and other users:
https://github.com/home-assistant/addons/issues/3594
```

**Comment by sakirsek** [comment]
I have this problem too. Still no solution?

**Comment by LeoCal** [comment]
AFAIK, unfortunately no solution yet :-(

**Comment by GunterAv** [comment]
Hello,

the same with HAOS: 12.4

I found an alternative solution (without VLC)
[https://smarterkram.de/6217/](https://github.com/home-assistant/addons/issues/url) (In German)
Also works with short mp3 files.

Best regards
Gunter

**Comment by LeoCal** [comment]
Yes, I also have an MPD based addon that works fine, but ideally this needs to be fixed in VLC addon as that's part of the Home Assistant "official add-ons".

**Comment by GunterAv** [comment]
Hello,

I'm right there with you.
But I need the sound output via the RasPi output and didn't want to wait any longer.

Greetings

**Comment by Grafton1046** [comment]
Throwing my hat into the ring, it's not just raspberry Pi's.
After the update 12.4 I'm not able to get any sound out of my setup, running baremetal with x86-64

**Comment by FadilAbdulla** [comment]
I also have the issue since I updated to 2024.5

**Comment by rodolfoleo** [comment]
I had the same problem on 12.3 so I downgraded to 12.2 and the audio was fixed. 

Anybody knows if OS 13 still presents this issue?

**Comment by GunterAv** [comment]
in 13.0 the proplem was NOT fixed.

**Comment by basvdijk** [comment]
I experience the same on my RPI4 and noticed a very high cpu usage when playing audio or TTS. This might cause the stuttering sound.

**Comment by bitbytie** [comment]
I'm also stuck on version 12.2 until this issue is resolved.  
Is there anything I can do to help diagnose the root cause?

**Comment by sairon** [comment]
I had an opportunity to look at this issue a bit more in depth, and I'm afraid I don't have good news. The problem indeed happens only for VLC, which, when ran with debug logging, shows it has issues with synchronization and logs `playback way too early ... playing silence` or `playback way too late ... playing silence` couple of times. Playing audio using other tools (like `ffplay`) is fine. I checked and tried adjusting buffer values but it has no effect.

I have confirmed that Raspberry Pi OS has exact the same issue **if PulseAudio and VLC** is used. The problem only happens on latest (for RPi) 6.6 kernel, 6.1 is fine - and 12.2 to 12.3 was right where we updated to Raspberry Pi's current kernel.

Unfortunately, I don't have capacity to pursue this issue, these are the key information that could be used to report the issue upstream in [raspberrypi/linux](https://github.com/raspberrypi/linux/) repo - install PulseAudio (it's not used as the default sound server though) on the latest RPi OS and try playing audio files with VLC. If the cause is found and resolved there, we can apply the fix to HAOS as well.

**Comment by ClubspeedChris** [comment]
Running the latest versions and all of my door/alarm sounds now just make weird .2 sec chirps. Really annoying. Is anyone looking at this, anything we can do to help?

**Comment by NoahJamesTech** [comment]
+1, this issue is driving me crazy. Is there anything we can do to help? And/or resolve the issue temporarily on our own hardware?

**Comment by mib1185** [comment]
As mentioned in https://github.com/home-assistant/operating-system/issues/3373#issuecomment-2297181620, reporting this issue to the [raspberrypi/linux](https://github.com/raspberrypi/linux/) repository by affected users would help to get it fixed upstream, thx :+1:

**Comment by robertodomotico** [comment]
I have the same problem with the output on VLC only when I use the piper voice or with some mp3 audio.
The strange thing is that if I use the google voice on the VLC output it works fine.
I don't know if this can give you any ideas but I hope it gets solved

**Comment by LeoCal** [comment]
In my case, instead, this is happening also using google TTS on the VLC output, on a Raspberry PI 4.

**Comment by basvdijk** [comment]
Anyone tested if the latest HAOS fixed this issue?

**Comment by NoahJamesTech** [comment]
I tested with the last major version (I think 14.0) and it still wasn't working. Reverted back to 12.2 where it was working.

**Comment by basvdijk** [comment]
@NoahJamesTech Thanks! Looks like we are stuck with 12 or need to find another solution. I have the feeling this is not going to be fixed with future HA updates.

**Comment by HiTechTLS** [comment]
Confirming this is still broken on 2025.3.1 & 14.2. Choppy, missing, and broken audio with any TTS used

**Comment by mib1185** [comment]
with HAOS 15 we're changing to kernel 6.12 for all systems, except the PI, since 6.12 is not final available (_but [it is near to be](https://forums.raspberrypi.com/viewtopic.php?t=379745)_).
I've done some tests with an rpi4 ...
I'm able to reproduce the issue with the current latest kernel `6.6.74+rpt-rpi-v8` by playing an web audio stream via pullseaudio with vlc and ensure some high load with `dd if=/dev/urandom bs=10M count=1000 of=/dev/null`
Then I've updated to the current test kernel `6.12.18-v8+` and tried to reproduce the issue again, but this time the audio was playing fine all the time, even when under high load 👍 

TL;DR: with upcoming kernel [6.12](https://forums.raspberrypi.com/viewtopic.php?t=379745) the issue should be gone

**Comment by basvdijk** [comment]
There is still no confirmation that it worked in the latest HA version. Curious to see if anyone can confirm it is fixed.

**Comment by mib1185** [comment]
the kernel will be updated to 6.12 also for the RasPI in HA-OS 16 (_#4036_). so with HA-OS 16 the issue will be fixed

**Comment by sairon** [comment]
OS 16.0.rc1 is out which contains the latest kernel, can you folks check how it behaves for you? I can hear couple of dropouts when the VLC stream starts but it's fine then, and load is not affecting it.

**Comment by tvenkataraman** [comment]
I have updated the OS 16.0.rc1 and Core 2025.6.3 on my RasPI 4. But I am still facing the issue that the chopping sound. 
Earlier I have used 32bit OS for many years on RasPI 3 and I never had such issue and since 32bit is deprecated, I am now updating my setup to 64bit.

So, I kindly request you to take a look and fix the issue please.
`Installation method
Home Assistant OS
Core
2025.6.3
Supervisor
2025.06.2
Operating System
16.0.rc1
Frontend
20250531.4`

**Comment by bitbytie** [comment]
> OS 16.0.rc1 is out which contains the latest kernel, can you folks check how it behaves for you? I can hear couple of dropouts when the VLC stream starts but it's fine then, and load is not affecting it.

I've updated the OS to 16.0.rc1, unfortunately I'm still experiencing the same behaviour with audio playback via the VLC add-on.  I'm away this weekend, but will try a fresh installation of HA on a spare Raspberry PI and report back next week.

**Comment by robertodomotico** [comment]
I don't know if my comment can be useful but I had that problem or a very similar one when I used a USB speaker connected directly to the RPI (which therefore probably absorbed too much current from the USB)
Since I added an audio interface (therefore with minimal absorption) in turn connected to a self-powered speaker the playback is fluid.

**Comment by basvdijk** [comment]
> > OS 16.0.rc1 is out which contains the latest kernel, can you folks check how it behaves for you? I can hear couple of dropouts when the VLC stream starts but it's fine then, and load is not affecting it.
> 
> I've updated the OS to 16.0.rc1, unfortunately I'm still experiencing the same behaviour with audio playback via the VLC add-on. I'm away this weekend, but will try a fresh installation of HA on a spare Raspberry PI and report back next week.

I was wondering if you had tried the fresh install :)

**Comment by bitbytie** [comment]
I've performed a clean installation of Home Assistant OS 16.0 on a Raspberry Pi 4. After that, I installed and enabled both the VLC add-on and the vlc-telnet device. I then uploaded a sample MP3 file (from https://pixabay.com/sound-effects/a320-tritone-chime-104562/).

Unfortunately, playback of this mp3 still suffers from dropouts, as experienced previously.

Edit - I'm using a self powered speaker and the official Raspberry PI power supply.

**Comment by robertodomotico** [comment]
Cosa hai collegato come altoparlante ?

Il dom 13 lug 2025, 15:30 Carl Boyes ***@***.***> ha scritto:

**Comment by bitbytie** [comment]
Speaker is a JBL Flip 2, connected directly to the audio jack on the Pi 4.

**Comment by robertodomotico** [comment]
Te l'ho chiesto perché ho sempre avuto problemi anche io con questa uscita
audio e alla fine ho risolto collegando un dongle audio USB al Raspberry e
da lì dentro un altoparlante amplificato.
Avevo provato anche con un altoparlante amplificato USB ma se alzavo troppo
il volume la porta USB andava in crisi.

La stessa identica cosa anche con minipc e anche lì ho usato il dongle USB
audio, mai più avuto problemi.
Non so se può esserti utile ma almeno risolvi per sempre.

Il dom 13 lug 2025, 15:38 Carl Boyes ***@***.***> ha scritto:

**Comment by itCarl** [comment]
Has there been any progress?

**Comment by jormalaaksonen** [comment]
I have the same issue with core-2025.10.2, Home Assistant OS 16.2, 6.12.34-haos-raspi, rpi4-64 and the speaker connected with the 3.5mm audio jack. 

I don't have experience on earlier versions before spring 2025 since when the problem has persisted in all releases.

I can report the additional fact that at least in my case the audio starts to play very well after approx 12 seconds of jittering. So I suspect it is somehow related to insufficient buffering.

**Comment by narendraj9** [comment]
> I can report the additional fact that at least in my case the audio starts to play very well after approx 12 seconds of jittering. So I suspect it is somehow related to insufficient buffering.

I have the same issue with the latest release (Home Assistant OS - 16.3).

**Comment by VISTALL** [comment]
Hello. Looks like I have same issue. Raspberry Pi 4 8 gb. Audio lagged if used VLC addon + VLC-Telnet integration
```
Installation method Home Assistant OS
Core 2025.12.1 
Supervisor 2025.12.2
Operating System 16.3
Frontend 20251203.1
```

**Comment by dkmatt0** [comment]
Same with Home Assistant OS 17.0

**Comment by dkmatt0** [comment]
How do you get audio? Are you still on 12.2?
I had to update because 12.2 is no longer supported and it's impossible to install new add-ons.

**Comment by basvdijk** [comment]
Today I had to update since my doorbell was not supported anymore in the old HA builds.

I am now running:

```
Core 2026.2.1
Supervisor 2026.01.1
Operating System 17.0
Frontend 20260128.6
```

VLC-TELNET on the builtin audio jack of my RPI4 was stuttering as expected.

I had an [USB Audioquest DragonFly Black audio device](https://www.audioquest.com/products/dragonfly-black) laying around and tested.

The hardest part for me was to find if the RPI supported the DragonFly and where to set it as audio device.

Settings -> App -> VLC -> Configuration

<img width="501" height="329" alt="Image" src="https://github.com/user-attachments/assets/191d53e2-ab89-4646-9079-95a726b0baaa" />

This worked like a charm. No more stuttering! 

So chances are that a cheap USB audio from AliExpress works as well. E.g. https://s.click.aliexpress.com/e/_c4Myxxvz

**Comment by rodolfoleo** [comment]
> The hardest part for me was to find if the RPI supported the DragonFly and where to set it as audio device.
> 
> Settings -> App -> VLC -> Configuration
> 
> So chances are that a cheap USB audio from AliExpress works as well. E.g. https://s.click.aliexpress.com/e/_c4Myxxvz

Thank you so much for this idea. I got the Ugreen usb sound card, updated everything in HA and upon restart I selected this new audio card as VLC output. Everything is working perfectly well! Excellent solution to a months-long problem.

**Comment by tvenkataraman** [comment]
Hello,
Tested on Home Assistant Core version 2026.5.4 and Operating System version 17.3 and the problem still present on Raspberry Pi 4.
I used a work around to use external VLC telnet integration to solve the Issue.  Thanks for your support.

```
Installation method - Home Assistant OS
Core - 2026.5.4
Supervisor - 2026.05.0
Operating System - 17.3
Frontend - 20260429.4
```
